modules/porous_flow/include/materials/PorousFlowMaterialPermeabilityFromPorosity.h

```diff
@@ -0,0 +1,76 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef POROUSFLOWMATERIALPERMEABILITYFROMPOROSITY_H
+#define POROUSFLOWMATERIALPERMEABILITYFROMPOROSITY_H
+
+#include "DerivativeMaterialInterface.h"
+#include "Material.h"
+
+#include "PorousFlowDictator.h"
+
+//Forward Declarations
+class PorousFlowMaterialPermeabilityFromPorosity;
+
+template<>
+InputParameters validParams<PorousFlowMaterialPermeabilityFromPorosity>();
+
+/**
+ * Material designed to provide the permeability tensor
+ * which is a function of porosity
+ */
+class PorousFlowMaterialPermeabilityFromPorosity : public DerivativeMaterialInterface<Material>
```

Comment:
I suggest deriving from PorousFlowPermeabilityConst (or creating a new PorousFlowPermeabilityUnity and subclassing the "Const" and "KozenyCarman" from it).  That will be more ordered.